internal/service/codebuild/fleet.go

```diff
@@ -712,7 +710,11 @@ func flattenScalingConfiguration(apiObject *types.ScalingConfigurationOutput) ma
 		tfMap["target_tracking_scaling_configs"] = flattenTargetTrackingScalingConfigs(v)
 	}
 
-	return tfMap
+	if len(tfMap) == 0 {
+		return nil
+	}
+
+	return []interface{}{tfMap}
```

Comment:
The Read operation returns an empty `ScalingConfigurationOutput` object after the scaling configuration is removed and this function flattened it to such state:
```
scaling_configuration {
}
```
Which caused a persistent diff.

I skimmed over other services and resources but I couldn't really find the standard way to deal with such empty nested blocks.

Reply:
This seems reasonable to me given the situation. If this is eventually serialized properly to a nil value in the AWS SDK the flattener will still do the "right" thing.